src/bodywork/workflow_execution.py

```diff
@@ -183,25 +181,36 @@ def _setup_namespace(config) -> str:
     :param config: Bodywork config.
     :return: Name of namespace.
     """
-    namespace = ""
+    namespace = (
+        config.project.namespace if config.project.namespace else config.project.name
+    )
     try:
-        if config.project.namespace:
-            namespace = config.project.namespace
-        else:
-            namespace = config.project.name
         if not k8s.namespace_exists(namespace):
             _log.info(f"Creating namespace: {namespace}")
             k8s.create_namespace(namespace)
         if not k8s.service_account_exists(namespace, BODYWORK_STAGES_SERVICE_ACCOUNT):
             _log.info(f"Creating service account: {BODYWORK_STAGES_SERVICE_ACCOUNT}")
-            k8s.setup_job_and_deployment_service_account(namespace)
+            k8s.setup_stages_service_account(namespace)
         return namespace
     except ApiException as e:
         raise BodyworkNamespaceError(
             f"Unable to setup namespace: {namespace} - {e}"
         ) from e
 
 
+def config_deploys_services(config: BodyworkConfig) -> bool:
```

Comment:
Maybe `workflow_deploys_services` is better?

Reply:
Maybe a simple test for this, too?

Reply:
The name was your suggestion. I prefer `workflow_deploys_services `. 

This is implicitly tested in `workflow_execution.py` in the last two tests.

Reply:
Yes, that's right - you test the tear-down!